CMakeLists.txt

```diff
@@ -128,14 +152,20 @@ endif()
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src)
 
-if(EXISTS ${PROJECT_SOURCE_DIR}/mshadow/cmake)
-  include(mshadow/cmake/mshadow.cmake)
+if(FIRST_CUDA)
```

Comment:
Can there be a more descriptive name other than “first CUDA”? What is it, really?

Reply:
the “first CUDA” mean "cmake first class language cuda".

Reply:
I don;t know what that means :(

Reply:
https://cmake.org/cmake/help/latest/release/3.8.html?highlight=cuda#cuda

Reply:
ok, can the macro be more descriptive of that? Like CUDA_AS_A_FIRST_CLASS_LANGUAGE or something like that?  "CUDA_FIRST" seems cryptic.